lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -84,64 +401,58 @@ function collectAllStringsInDir(dir, strings = {}) {
           let lastPropertyEndIndex = 0;
           for (const property of stmt.declarations[0].init.properties) {
             const key = property.key.name;
-            const message = exportVars.UIStrings[key];
-            const description = computeDescription(ast, property, lastPropertyEndIndex);
-            strings[`${relativePath} | ${key}`] = {message, description};
-            lastPropertyEndIndex = property.range[1];
-          }
-        }
-      }
-    }
-  }
+            const val = exportVars.UIStrings[key];
+            const {description, examples} = computeDescription(ast, property, val, lastPropertyEndIndex);
 
-  return strings;
-}
+            const converted = convertMessageToPlaceholders(val, examples);
 
```

Comment:
should we add a sanity check function to the converted response? A lot of things are already checked in each transform, but we don't have a great place for checking e.g. the completed string has an even number of `$` placeholders and and an equal number of matching `{` and `}`

Reply:
what do you think of this?

Reply:
Talked in person, I like the idea of _basic_ sanity checks.  Added the first one.